lib/ecto_adapters_dynamodb/query.ex

```diff
@@ -46,8 +48,19 @@ defmodule Ecto.Adapters.DynamoDB.Query do
           end
       parsed -> parsed
     end
-    do_fetch_recursive = fn(qry) ->
-      fetch_recursive(&ExAws.Dynamo.query/2, table, qry, parse_recursive_option(:query, opts), %{})
+    do_fetch_recursive = fn(query) ->
+      fetch_recursive(&ExAws.Dynamo.query/2, table, query, parse_recursive_option(:query, opts), %{})
+    end
+    do_in_query = fn(query, hash_values) ->
+      response_map = %{@responses_element_key => %{table => []}}
+
+      Enum.reduce(hash_values, response_map, fn(hash_value, acc) ->
+        # When receiving a list of values to query on, construct a custom query for each of those values to pass into do_fetch_recursive/1.
+        %{"Items" => items} = Kernel.put_in(query, [:expression_attribute_values, :hash_key], hash_value)
```

Comment:
`Kernel` is a magic namespace which is always imported - you don't need to specify it on calls to its functions. In other words, you can just call `put_in` rather than `Kernel.put_in`.

Reply:
Yeah, I'm familiar with that - I just opted to be explicit. It seems like this overall logic may get revisited - I'd consider just going with the more customary `put_in`.